forest/benchmarking/direct_fidelity_estimation.py

```diff
@@ -112,10 +113,24 @@ def generate_exhaustive_process_dfe_experiment(benchmarker: BenchmarkConnection,
         Note that we assume qubits are initialized to the ``|0>`` state.
     :return: an ObservablesExperiment that constitutes a process DFE experiment.
     """
-    expt = ObservablesExperiment(
-        list(_exhaustive_dfe(benchmarker=benchmarker, program=program, qubits=qubits,
-                             in_states=[None, plusX, minusX, plusY, minusY, plusZ, minusZ])),
-        program=program)
+    settings = []
+    for pauli_labels in [''.join(x) for x in itertools.product('IXYZ', repeat=len(qubits))][1:]:
+        observable = benchmarker.apply_clifford_to_pauli(program,
+                                                         str_to_pauli_term(pauli_labels, qubits))
+        non_identity_idx = [0 if label == 'I' else 1 for label in pauli_labels]
+        state_labels = ['Z' if label == 'I' else label for label in pauli_labels]
+        for eigenstate in itertools.product([0, 1], repeat=len(qubits)):
+            in_state = functools.reduce(mul,
```

Comment:
this assignment can be simplified by directly passing the generator to `TensorProductState`'s initializer

```suggestion
            in_state = TensorProductState(_OneQState(l, s, q) for l,s,q in zip(state_labels, eigenstate, qubits))
```